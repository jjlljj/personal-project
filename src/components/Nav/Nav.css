.nav-header {
  width: 100%;
  min-width: 320px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #d2d2d2;
}

.nav-header nav {
  display: flex;
}

.nav-header h1 {
  margin: 20px; 
}

.nav-header h1 a span {
  color: #003780f0;
}

.nav-header h1 a  {
  color: #0000ffcf;
}

.nav-header nav div {
  margin: 20px;
}

.nav-header .active {
  color: blue;
}

@media screen and ( max-width:500px){
  .nav-header {
    justify-content: center;
    flex-wrap: wrap;
  }

  .nav-header h1 {
    font-size: 36px;
    margin: 30px auto 0;
  }
}
