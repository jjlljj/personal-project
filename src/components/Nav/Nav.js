import React from 'react'
import { NavLink, Link } from 'react-router-dom'
import './Nav.css'

const Nav = () => {
  return (
    <header className="nav-header">
      <h1>
        <Link to="/">
          toneDetector
        </Link>
      </h1>
      <nav>
        <div>
          <NavLink to="/analyze">
            Analyze
          </NavLink>
        </div>
        <div>
          <NavLink to="/about">
            About
          </NavLink>
        </div>
      </nav>
    </header>
  )
}

export default Nav