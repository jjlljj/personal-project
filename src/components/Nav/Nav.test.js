import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

describe('Nav', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<Nav />)
  })

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
