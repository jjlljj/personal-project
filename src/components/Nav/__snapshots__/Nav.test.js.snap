// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/"
          >
                    SITE HEADER
          </NavLink>
</h1>,
        <nav>
          <div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/"
                    >
                              Linkto
                    </NavLink>
          </div>
          <div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/"
                    >
                              Linkto
                    </NavLink>
          </div>
</nav>,
      ],
      "className": "nav-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            SITE HEADER
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "SITE HEADER",
            "to": "/",
          },
          "ref": null,
          "rendered": "SITE HEADER",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/"
              >
                            Linkto
              </NavLink>
</div>,
            <div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/"
              >
                            Linkto
              </NavLink>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
>
                Linkto
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Linkto",
                "to": "/",
              },
              "ref": null,
              "rendered": "Linkto",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
>
                Linkto
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Linkto",
                "to": "/",
              },
              "ref": null,
              "rendered": "Linkto",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/"
            >
                        SITE HEADER
            </NavLink>
</h1>,
          <nav>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/"
                        >
                                    Linkto
                        </NavLink>
            </div>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/"
                        >
                                    Linkto
                        </NavLink>
            </div>
</nav>,
        ],
        "className": "nav-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/"
>
              SITE HEADER
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "SITE HEADER",
              "to": "/",
            },
            "ref": null,
            "rendered": "SITE HEADER",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/"
                >
                                Linkto
                </NavLink>
</div>,
              <div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/"
                >
                                Linkto
                </NavLink>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/"
>
                  Linkto
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Linkto",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Linkto",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/"
>
                  Linkto
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Linkto",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Linkto",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
